pkg/infra/httpclient/httpclientprovider/http_client_provider.go

```diff
@@ -29,5 +33,23 @@ func New(cfg *setting.Cfg) httpclient.Provider {
 
 	return newProviderFunc(sdkhttpclient.ProviderOptions{
 		Middlewares: middlewares,
+		ConfigureTransport: func(opts sdkhttpclient.Options, transport *http.Transport) {
+			newConntrackRoundTripper(connName)
+		},
 	})
 }
+
+// newConntrackRoundTripper takes a http.DefaultTransport and adds the Conntrack Dialer
+// so we can instrument outbound connections
+func newConntrackRoundTripper(name string) *http.Transport {
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+	transport.DialContext = conntrack.NewDialContextFunc(
+		conntrack.DialWithTracing(),
+		conntrack.DialWithName(name),
```

Comment:
How are the name used? Should probably use something similar like this? https://github.com/grafana/grafana/blob/ce513e4637e2f8fbe56bb8a8c9077bf57a798ca2/pkg/infra/httpclient/httpclientprovider/datasource_metrics_middleware.go#L65